cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -171,6 +170,15 @@ static std::shared_ptr<StructScalar> ColumnChunkStatisticsAsStructScalar(
     return nullptr;
   }
 
+  auto maybe_min = min->CastTo(field->type());
+  auto maybe_max = max->CastTo(field->type());
```

Comment:
Does this change behaviour? For a dictionary with string values, is field->type() string or dictionary?

Reply:
`StatisticsAsScalars` returns scalars whose types are the correct *physical type*, so even if the column was `dictionary(string)` min and max would be just `string` before this cast

Reply:
(IE, it only changes behavior in cases where the physical type wasn't appropriate)